tests/foreman/api/test_user.py

```diff
@@ -372,3 +372,168 @@ def test_positive_update(self):
                 # pylint:disable=no-member
                 set([role.id for role in chosen_roles] + [self.anon_role.id]),
             )
+
+
+class SshKeyInUser(APITestCase):
+    """Implements the SSH Key in User Tests"""
+
+    @stubbed
+    @tier1
+    def test_positive_create_ssh_key(self):
+        """SSH Key can be added to User
+
+        :id: d00905f6-3a70-4e2f-a5ae-fcac18274bb7
+
+        :steps:
+
+            1. Create new user with all the details
+            2. Add SSH key to the above user
+
+        :expectedresults: SSH key should be added to user
+        """
+
+    @stubbed
+    @tier1
+    def test_positive_create_ssh_key_super_admin(self):
+        """SSH Key can be added to Super Admin user
+
+        :id: 397eea22-759c-4cd4-bda1-0e7835566c72
+
+        :steps:
+
+            1. Add SSH Key to the Super Admin user
+
+        :expectedresults: SSH Key should be added to Super Admin user
+        """
+
+    @stubbed
+    @tier1
+    def test_negative_create_ssh_key(self):
+        """Invalid ssh key can not be added in User Template
+
+        :id: e924ff03-8b2c-4ab9-a054-ea491413e143
+
+        :steps:
+
+            1. Create new user with all the details
+            2. Attempt to add invalid string as SSH Key to above user
+                e.g blabla
+
+        :expectedresults: Invalid SSH key should not be added in user
+        """
+
+    @stubbed
+    @tier1
+    def test_positive_error_create_invalid_ssh_key(self):
+        """Satellite returns exception on invalid ssh key creation in user
+
+        :id: 3b2c37bd-4f95-44bc-8e57-620029535355
+
+        :steps:
+
+            1. Create new user with all the details
+            2. Attempt to add invalid string as SSH Key to above user
+                e.g blabla
+
+        :expectedresults: Satellite returns 'Fingerprint could not be
+            generated' error
+        """
+
+    @stubbed
+    @tier1
+    def test_positive_error_create_invalid_length_ssh_key(self):
```

Comment:
it should be a negative test case and you should state in the description not the error as main goal of the case, but the fact itself, like:
```
Attempt to add SSH key that has invalid length
```
and in expcted result:
```
error is raised
```